L1Trigger/CSCTriggerPrimitives/src/CSCUpgradeMotherboard.h

```diff
@@ -0,0 +1,128 @@
+#ifndef L1Trigger_CSCTriggerPrimitives_CSCUpgradeMotherboard_h
+#define L1Trigger_CSCTriggerPrimitives_CSCUpgradeMotherboard_h
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "L1Trigger/CSCTriggerPrimitives/src/CSCMotherboard.h"
+#include "L1Trigger/CSCTriggerPrimitives/src/CSCUpgradeMotherboardLUT.h"
+#include "L1Trigger/CSCTriggerPrimitives/src/CSCUpgradeMotherboardLUTGenerator.h"
+#include <boost/variant.hpp>
+
+// generic container type
+template <class T>
+using match = std::pair<unsigned int, T>;
+
+template <class T>
+using matches = std::vector<std::pair<unsigned int, T> >;
+
+template <class T>
+using matchesBX = std::map<int, std::vector<std::pair<unsigned int, T> > >;
+
+class CSCGeometry;
+class CSCChamber;
+
+class CSCUpgradeMotherboard : public CSCMotherboard
+{
+public:
+
+  /** for the case when more than 2 LCTs/BX are allowed;
+      maximum match window = 15 */
+  class LCTContainer {
+  public:
+    LCTContainer (unsigned int match_trig_window_size ) : match_trig_window_size(match_trig_window_size){}
+    CSCCorrelatedLCTDigi& operator()(int bx, int match_bx, int lct) { return data[bx][match_bx][lct]; }
+    std::vector<CSCCorrelatedLCTDigi> getTimeMatched(const int bx) const;
+    std::vector<CSCCorrelatedLCTDigi> getMatched() const;
+    CSCCorrelatedLCTDigi data[CSCMotherboard::MAX_LCT_BINS][15][2];
+    const unsigned int match_trig_window_size;
+  };
+
+  CSCUpgradeMotherboard(unsigned endcap, unsigned station, unsigned sector,
+                    unsigned subsector, unsigned chamber,
+                    const edm::ParameterSet& conf);
+
+   //Default constructor for testing
+  CSCUpgradeMotherboard();
+
+  virtual ~CSCUpgradeMotherboard();
+
+  template <class S>
+  bool compare(const S& p, const S& q);
+
+  template <class S>
+  S intersection(const S& d1, const S& d2);
+
+  /** Methods to sort the LCTs */
+  static bool sortLCTsByQuality(const CSCCorrelatedLCTDigi&, const CSCCorrelatedLCTDigi&); 
+  static bool sortLCTsByGEMDphi(const CSCCorrelatedLCTDigi&, const CSCCorrelatedLCTDigi&); 
+  void sortLCTs(std::vector<CSCCorrelatedLCTDigi>& lcts, bool (*sorter)(const CSCCorrelatedLCTDigi&,const CSCCorrelatedLCTDigi&));
+
+  void setCSCGeometry(const CSCGeometry *g) { csc_g = g; }
+  void setupGeometry();
+  void debugLUTs();
+
+ protected:
+
+  /** Chamber id (trigger-type labels). */
+  unsigned theRegion;
+  unsigned theChamber;
+  Parity par;
+
+  edm::ParameterSet tmbParams_;
```

Comment:
I plan to use the tmbParams_ in the constuctors of the derived classes. In principle I could construct a pset object in the constructor of each derived class. Whichever is best I suppose.